core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -362,7 +362,7 @@ oppia.controller('StateInteraction', [
         var imageWithRegions = currentCustomizationArgs.imageAndRegions.value;
         for (var j = 0; j < imageWithRegions.labeledRegions.length; j++) {
           _answerChoices.push({
-            val: imageWithRegions.labeledRegions[j].label,
+            val: j,
```

Comment:
Yeah, you're right. This will break the previous explorations. I tried something like this also but the problem is that it breaks other interactions like Multiple Choice interaction whose value is equal to the actual text - something like `<p>a</p>`. In that case, id is the parameter that works. A solution likes this handles both perfectly but I fear it might be a bit hacky - 
```
      $scope.getSelectionIndex = function() {
        for (var i = 0; i < $scope.options.length; i++) {
          if (($scope.options[i].val === $scope.selection) || ($scope.options[i].id === $scope.selection)) {
            return i;
          }
        }
      };
```